contracts/peripheral/YieldDirectorV2.sol

```diff
@@ -0,0 +1,485 @@
+        @notice Redeem recipient's full donated amount of sOHM at current index as gOHM
+    */
+    function redeemAllYieldAsSohm() external {
+        uint256 amountRedeemed = _redeemAll();
+
+        staking.unwrap(msg.sender, amountRedeemed);
+    }
+
+    /************************
+     * Internal Functions
+     ************************/
+
+    /**
+        @notice Creates a new deposit directing the yield from the deposited gOHM amount
+                to the prescribed recipient
+        @param amount_ Quantity of gOHM deposited redirecting yield to the recipient
+        @param recipient_ The address of the user who will be entitled to claim the donated yield
+    */
+    function _createDeposit(uint256 amount_, address recipient_) internal returns (uint256 depositId) {
+        if (isInvalidDeposit(amount_, recipient_)) revert YieldDirector_InvalidDeposit();
+
+        depositId = _deposit(msg.sender, amount_);
+        recipientIds[recipient_].push(depositId);
+        recipientLookup[depositId] = recipient_;
+
+        emit Deposited(msg.sender, recipient_, amount_);
+    }
+
+    /**
+        @notice Increases the amount of gOHM directing yield to a recipient
+        @param depositId_ The global ID number of the deposit to add the additional deposit to
+        @param amount_ Quantity of new gOHM deposited redirecting yield to the current deposit's recipient
+    */
+    function _increaseDeposit(uint256 depositId_, uint256 amount_) internal {
+        if (isInvalidUpdate(depositId_, amount_)) revert YieldDirector_InvalidUpdate();
+        if (depositInfo[depositId_].depositor != msg.sender) revert YieldDirector_NotYourDeposit();
+
+        _addToDeposit(depositId_, amount_);
+
+        emit Deposited(depositInfo[depositId_].depositor, recipientLookup[depositId_], amount_);
+    }
+
+    /**
+        @notice Withdraw gOHM deposit from vault
+        @param depositId_ Deposit ID to remove gOHM deposit from
+        @param amount_ Amount of gOHM deposit to remove and return to donor 
+    */
+    function _withdraw(uint256 depositId_, uint256 amount_) internal returns (uint256 amountWithdrawn) {
+        if (isInvalidWithdrawal(amount_)) revert YieldDirector_InvalidWithdrawal();
+
+        if (amount_ >= _toAgnostic(depositInfo[depositId_].principalAmount)) {
+            amountWithdrawn = _withdrawAllPrincipal(depositId_);
+        } else {
+            _withdrawPrincipal(depositId_, amount_);
+            amountWithdrawn = amount_;
+        }
+
+        emit Withdrawn(msg.sender, recipientLookup[depositId_], amountWithdrawn);
+    }
+
+    /**
+        @notice Redeem available gOHM yield from a specific deposit
+        @param depositId_ Deposit ID to withdraw gOHM yield from
+    */
+    function _redeem(uint256 depositId_) internal returns (uint256 amountRedeemed) {
+        if (redeemDisabled) revert YieldDirector_RedeemsDisabled();
+        if (recipientLookup[depositId_] != msg.sender) revert YieldDirector_NotYourDeposit();
```

Comment:
Should rename this error to `YieldDirector_NotYourYield` since NotYourDeposit is covered in the new YieldSplitter